### PR TITLE
add setOuterHtml for HtmlNode

### PR DESCRIPTION
**Pull request by 0xuhe**

### CODE REVIEW

**Comment by plonknimbuzz** [comment]
why we need to set outerHtml?

**Comment by 0xuhe** [comment]
Maybe we need replace node sometimes, when our content from user submit
eg: 
```php

$node = '<span data-link='http://foo.com'>foo.com</span>';

$dom->load($node);
$dom->setOuterHtml('<a href="http://foo.com">foo.com</a>');

$model->body = (string) $dom;
$model->save();

```

refer to [link](https://github.com/paquettg/php-html-parser/issues/125#issuecomment-366585412)

File: composer.json [hunk 1]
```diff
@@ -1,5 +1,5 @@
 {
-    "name": "thesoftwarefanatics/php-html-parser",
+    "name": "gfd6th/php-html-parser",
```
Comment (alcaeus) [inline composer.json]: This needs to be removed, can't merge the pull request with it.

**Review by alcaeus — CHANGES_REQUESTED** [review]
Thanks for your contribution. Please add tests for your changes and fix review issues.

**Comment by plonknimbuzz** [comment]
@gfd6th  
i see. i think this will be great for the future.
i ask that question, because i never need/use that method.  but maybe someday this feature will help me alot.
thanks for the explaination, very usefull for me.

**Comment by alcaeus** [comment]
After taking another look at the functionality, we will not be merging this. The outerHtml property is generated and should thus be considered immutable. Changing the outerHtml of a node should happen by creating a new node and adding all children from the original node. Thank you for your contribution.

**Pull request closed without merging**